### PR TITLE
Initial stab at mass scaling for explicit dynamics 

### LINKED ISSUE #22997: Mass scaling for explicit dynamics
## Reason
Explicit dynamics simulations require a time step size smaller than the critical time step. This requirement determines to a great extent the speed of the simulation. One numerical treatment to allow for larger time step sizes is to selectively add mass to some elements to increase the overall simulation time step size (since the critical time step of the simulation is the _smallest_ element critical time step size). For this reason, the advanced user may want to add mass to the finite elements to obtain the results faster. Obviously, care is required when employing mass scaling since it can potentially change the fundamental dynamics of the system. 

## Design
Usage of a material property to determine the amount of mass/density required to achieve the user-selected time step.

## Impact
No impact on existing capabilities.

### PR DESCRIPTION
**Pull request by recuero**
Closes #22997.

### CODE REVIEW

File: modules/tensor_mechanics/include/postprocessors/CriticalTimeStep.h [hunk 1]
```diff
@@ -39,6 +39,9 @@ class CriticalTimeStep : public ElementPostprocessor, public GuaranteeConsumer
   /// Density of the material
   const MaterialProperty<Real> & _material_density;
 
+  /// Density of the material
+  const MaterialProperty<Real> * _density_scaling;
```
Comment (dschwen) [inline modules/tensor_mechanics/include/postprocessors/CriticalTimeStep.h:43]: `_scaled_material_density`? The current name makes it sound like a scaling factor

Comment (dschwen) [reply modules/tensor_mechanics/include/postprocessors/CriticalTimeStep.h:43]: Hm, I see, this is only the additional density.

File: modules/tensor_mechanics/src/materials/DensityScaling.C [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+DensityScaling::DensityScaling(const InputParameters & parameters)
+  : Material(parameters),
+    _desired_time_step(getParam<Real>("desired_time_step")),
+    _density_scaling(declareProperty<Real>("density_scaling")),
+    _material_density(getMaterialPropertyByName<Real>("density")),
+    _effective_stiffness(getMaterialPropertyByName<Real>("effective_stiffness")),
+    _factor(getParam<Real>("factor"))
+{
+  mooseDoOnce(mooseInfo("Usage of density (mass) scaling is only recommended for advanced users, "
```
Comment (dschwen) [inline modules/tensor_mechanics/src/materials/DensityScaling.C:41]: mooseInfo is already only outputting once

**Review by dschwen — DISMISSED** [review]

File: modules/tensor_mechanics/src/materials/DensityScaling.C (hunk 2 above)
Comment (recuero) [reply modules/tensor_mechanics/src/materials/DensityScaling.C]: Good to know. Made the update.

**Comment by recuero** [comment]
@dschwen Can you re-approve if everything is okay?

**Review by dschwen — APPROVED** [review]

**Pull request merged**